src/components/Snippet/index.tsx

```diff
@@ -0,0 +1,219 @@
+import styled from '@emotion/styled'
+import { Children, ComponentProps, useState } from 'react'
+import CopyButton from '../CopyButton'
+import Icon from '../Icon'
+import Stack from '../Stack'
+import Text from '../Text'
+
+const LINES_BREAK_REGEX = /\r\n|\r|\n/
+
+type Prefixes = 'lines' | 'command'
+
+const PreText = styled(Text, {
+  shouldForwardProp: prop =>
+    ![
+      'multiline',
+      'hasShowMoreButton',
+      'showMore',
+      'maxHeightExtended',
+    ].includes(prop),
+})<{
+  multiline?: boolean
+  hasShowMoreButton?: boolean
+  showMore?: boolean
+  maxHeightExtended: number
+}>`
+  margin: 0;
+  padding: ${({ theme }) => theme.space['2']};
+  padding-right: ${({ theme }) => theme.space['9']};
+  overflow-x: ${({ hasShowMoreButton, showMore }) =>
+    hasShowMoreButton && !showMore ? 'hidden' : 'auto'};
+  overflow-y: hidden;
+  ${({ multiline }) => (!multiline ? 'white-space: nowrap;' : '')}
+  height: auto;
+  max-height: ${({ multiline, showMore, maxHeightExtended }) =>
+    multiline && !showMore ? '120' : maxHeightExtended}px;
+  transition: max-height 300ms ease-in-out;
+  counter-reset: section;
+`
+
+const StyledSpan = styled('span', {
+  shouldForwardProp: prop =>
+    !['linePrefix', 'multiline', 'prefix'].includes(prop),
+})<{ linePrefix?: string; multiline?: boolean; prefix?: Prefixes }>`
+  display: block;
+
+  &:after {
+    content: '';
+    ${({ theme, multiline }) =>
+      multiline
+        ? `padding: ${theme.space['4']}`
+        : `padding-right: ${theme.space['8']}`};
+  }
+
+  &:before {
+    color: ${({ theme }) => theme.colors.neutral.textWeak};
+    width: ${({ prefix }) => (prefix === 'lines' ? '35px' : '')};
+    display: inline-flex;
+    justify-content: flex-end;
+    counter-increment: section;
+    content: ${({ prefix }) =>
+      prefix === 'lines' ? 'counter(section)' : "'$'"};
+    padding-right: ${({ theme }) => theme.space['1']};
+  }
+`
+
+const Container = styled('div', {
+  shouldForwardProp: prop => !['multiline'].includes(prop),
+})<{ multiline?: boolean }>`
+  position: relative;
+  display: flex;
+  justify-content: start;
+  max-width: 100%;
+  ${({ multiline }) => (multiline ? 'width: 100%;' : '')}
+  background: ${({ theme }) => theme.colors.neutral.backgroundStrong};
+  border-radius: ${({ theme }) => theme.radii.default};
+`
+
+const StyledStack = styled(Stack)`
+  width: 100%;
+`
+
+const ButtonContainer = styled('div', {
+  shouldForwardProp: prop => !['multiline'].includes(prop),
+})<{ multiline?: boolean }>`
+  position: absolute;
+  top: 0;
+  right: 0;
+  padding: ${({ theme, multiline }) =>
+    `${theme.space[multiline ? '2' : '1']} ${theme.space['2']} 0 0`};
+  background: ${({ theme }) => theme.colors.neutral.backgroundStrong};
+  border-radius: ${({ theme }) => theme.radii.default};
+  border: 2px solid transparent;
+  ${({ multiline, theme }) =>
+    !multiline
+      ? `box-shadow: -27px 0 19px -11px ${theme.colors.neutral.backgroundStrong}`
```

Comment:
No theme shadow ?

Reply:
@chambo-e same as said before as it is not generic shadow we cannot add it to theme